test/server/dataType.test.ts

```diff
@@ -0,0 +1,188 @@
+      const contentType = "text/plain";
+      const result = await toDataType(trigger, { data, contentType });
+      
+      expect(result.trigger).toBe(trigger);
+      expect(result.contentType).toBe(contentType);
+    });
+    
+    it("should throw error for invalid data type", async () => {
+      await expect(toDataType(trigger, { 
+        data: 123 as unknown as Buffer
+      })).rejects.toThrow("Invalid data type");
+    });
+  });
+  
+  describe("fromDataType", () => {
+    it("should handle null or undefined data", async () => {
+      const result = await fromDataType(null);
+      
+      expect(result.data).toBeDefined();
+    });
+    
+    it("should handle string data", async () => {
+      const data = "Hello, world!";
+      const result = await fromDataType(data);
+      
+      expect(result.data).toBeDefined();
+    });
+    
+    it("should handle object with contentType", async () => {
+      const jsonData = { message: "Hello" };
+      const data = {
+        contentType: "application/json",
+        base64: Buffer.from(JSON.stringify(jsonData)).toString("base64")
+      };
+      const result = await fromDataType(data);
+      
+      expect(result.data).toBeDefined();
+    });
+    
+    it("should handle ArrayBuffer data", async () => {
+      const text = "Hello, world!";
+      const data = new TextEncoder().encode(text).buffer as ArrayBuffer;
+      const result = await fromDataType(data);
+      
+      expect(result.data).toBeDefined();
+    });
+    
+    it("should handle Buffer data", async () => {
+      const text = "Hello, world!";
+      const data = Buffer.from(text);
+      const result = await fromDataType(data);
+      
+      expect(result.data).toBeDefined();
+    });
+    
+    it("should handle Uint8Array data", async () => {
+      const text = "Hello, world!";
+      const data = new TextEncoder().encode(text);
+      const result = await fromDataType(data);
+      
+      expect(result.data).toBeDefined();
+    });
+    
+    it("should handle plain object data", async () => {
+      const data = { message: "Hello, world!" };
+      const result = await fromDataType(data);
+      
+      expect(result.data).toBeDefined();
+    });
+    
+    it("should include metadata when provided", async () => {
+      const data = "Hello, world!";
+      const metadata: JsonObject = { key: "value" };
+      const result = await fromDataType(data, undefined, metadata);
+      
+      expect(result.data).toBeDefined();
+      expect(result.metadata).toEqual(metadata);
+    });
+    
+    it("should use provided contentType when specified", async () => {
+      const data = Buffer.from("Hello, world!");
+      const contentType = "text/plain";
+      const result = await fromDataType(data, contentType);
+      
+      expect(result.data).toBeDefined();
+    });
+    
+    it("should handle invalid data type gracefully", async () => {
+      const invalidData = 123;
```

Comment:
maybe use a function as invalid data here? if it doesn't throw i think the test case should fail